### PR TITLE
[EWPP-1539] EWPP-1756: Theming of Event Speaker field.

### PR DESCRIPTION
**Pull request by sergepavle**
### Description

[Insert description here]

### Change log

- Added:
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: tests/src/Functional/ContentEventRenderTest.php [hunk 1]
```diff
@@ -741,6 +766,76 @@ public function testEventRendering(): void {
...
+    $event_speaker = EventSpeaker::create([
+      'oe_event_role' => 'event role 1',
+      'type' => 'oe_default',
+      'oe_person' => $person,
+    ]);
+    $event_speaker->save();
+
+    $node->set('oe_event_speakers', [$event_speaker]);
+    $node->save();
+
```
Comment (22Alexandra) [inline tests/src/Functional/ContentEventRenderTest.php:806]: Could document the code a bit.

**Review by 22Alexandra — DISMISSED** [review]

File: oe_theme.theme [hunk 2]
```diff
@@ -1704,3 +1706,57 @@ function oe_theme_preprocess_oe_theme_helper_site_navigation(&$variables) {
...
+
+    // Make sure that Person entity is present.
+    if ($event_speaker->get('oe_person')->isEmpty() || !$event_speaker->get('oe_person')->entity instanceof NodeInterface) {
+      unset($variables['items'][$key]);
+      continue;
+    }
+    $person = $event_speaker->get('oe_person')->entity;
+    $cacheability->addCacheableDependency($person);
+
+    // Set portrait image URL.
```
Comment (nagyad) [inline oe_theme.theme:1734]: Remove this empty line

File: oe_theme.theme [hunk 3]
```diff
@@ -1704,3 +1706,57 @@ function oe_theme_preprocess_oe_theme_helper_site_navigation(&$variables) {
...
+    if (!$person->get('oe_person_photo')->isEmpty()) {
+
+      $renderable = $node_view_builder->viewField($person->get('oe_person_photo'), [
```
Comment (nagyad) [inline oe_theme.theme]: Variable name is too ambiguous.

File: oe_theme.theme [hunk 4]
```diff
@@ -1704,3 +1706,57 @@ function oe_theme_preprocess_oe_theme_helper_site_navigation(&$variables) {
...
+        'type' => 'oe_theme_helper_media_thumbnail_url',
+        'settings' => [
+          'image_style' => 'oe_theme_medium_no_crop',
+        ],
+      ]);
+      $cacheability->addCacheableDependency(CacheableMetadata::createFromRenderArray($renderable));
+      $item['content']['portrait_image'] = $renderable[0];
+    }
+    else {
+      $item['content']['portrait_image']['#markup'] = base_path() . drupal_get_path('theme', 'oe_theme') . '/images/user_icon.svg';
```
Comment (nagyad) [inline oe_theme.theme]: Normally we have a helper function to retrieve person image, no? If not shall we use one that incorporates these logic?

File: templates/field/field--entity-reference-revisions--oe-event-speaker.html.twig [hunk 5]
```diff
@@ -0,0 +1,25 @@
+{#
+/**
+ * @file
+ * Theme override for the entity reference revisions field wich allowed to use oe_event_speaker entity type.
+ *
+ * @see ./templates/field/field--bare.html.twig
+ */
+#}
+<h2 class="ecl-u-type-heading-2 ecl-u-mt-2xl ecl-u-mt-m-3xl ecl-u-mb-l">{{ label }}</h2>
```
Comment (nagyad) [inline templates/field/field--entity-reference-revisions--oe-event-speaker.html.twig:10]: I think we should use spaceless in this template.

File: tests/src/Functional/ContentEventRenderTest.php [hunk 6]
```diff
@@ -741,6 +766,76 @@ public function testEventRendering(): void {
...
+    $speakers = $this->assertSession()->elementExists('css', '.ecl-row.field-oe-event-speakers');
+    $speakers_items = $speakers->findAll('css', '.ecl-u-d-flex.ecl-u-pv-m.ecl-u-border-bottom.ecl-u-border-color-grey-15.ecl-col-12.ecl-col-m-6.ecl-col-l-4');
+    $this->assertCount(2, $speakers_items);
+    $portrait = $this->assertSession()->elementExists('css', '.ecl-u-flex-shrink-0.ecl-u-mr-s.ecl-u-media-a-s.ecl-u-media-bg-size-contain.ecl-u-media-bg-repeat-none', $speakers_items[0]);
+    $this->assertStringContainsString('oe_theme/images/user_icon.svg', $portrait->getAttribute('style'));
+    $meta = $this->assertSession()->elementExists('css', '.ecl-content-item__meta.ecl-u-type-s.ecl-u-type-color-grey-75.ecl-u-mb-xs.ecl-u-type-uppercase', $speakers_items[0]);
+    $this->assertEquals('event role 1', $meta->getText());
+    $link = $this->assertSession()->elementExists('css', '.ecl-link.ecl-link--standalone.ecl-u-type-bold.ecl-u-type-uppercase', $speakers_items[0]);
+    $this->assertStringContainsString($person->toUrl()->toString(), $link->getAttribute('href'));
+    $this->assertEquals($person->label(), $link->getText());
```
Comment (nagyad) [inline tests/src/Functional/ContentEventRenderTest.php:829]: I noticed that the person name is capitalized.
image

**Review by nagyad — CHANGES_REQUESTED** [review]

File: oe_theme.theme (hunk 4 above)
Comment (sergepavle) [reply oe_theme.theme]: Good point. I've reworked the existing helper function.

File: templates/field/field--entity-reference-revisions--oe-event-speaker.html.twig [hunk 7]
```diff
@@ -0,0 +1,27 @@
+{% spaceless %}
```
Comment (nagyad) [inline templates/field/field--entity-reference-revisions--oe-event-speaker.html.twig:1]: Move this above the `<h2>` line.

**Review by nagyad — CHANGES_REQUESTED** [review]

**Review by nagyad — APPROVED** [review]

**Review by 22Alexandra — APPROVED** [review]

**Pull request merged**